File: Modules/Kit/SAPS/SIE.Web.Kit.APS/EngineerPlans/Commands/GenerateEngineerPlanCommand.cs
using SIE.Kit.APS.EngineerPlans;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.Kit.APS.EngineerPlans.Commands
{
    /// <summary>
    /// 同步工程计划信息
    /// </summary>
    [JsCommand("SIE.Web.Kit.APS.EngineerPlans.Commands.GenerateEngineerPlanCommand")]
    public class GenerateEngineerPlanCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return RT.Service.Resolve<EngineerPlanController>().GenerateEngineerPlan();
        }
    }
}

File: Modules/Kit/SAPS/SIE.Web.Kit.APS/EngineerPlans/Commands/GenerateEngineerPlanCommand.js
SIE.defineCommand('SIE.Web.Kit.APS.EngineerPlans.Commands.GenerateEngineerPlanCommand', {
    meta: { text: "同步工程计划", group: "edit", iconCls: "icon-EditEntity icon-blue" },
    execute: function (view, source) {
        SIE.Msg.askQuestion('是否确认同步工程计划'.t(), function () {
            SIE.Msg.wait("正在同步工程计划中，请稍等...".t());
            view.execute({
                data: [],
                withIds: true,
                success: function (res) {
                    if (res.Success) {
                        var tempMsg = res.Result.replace(/\n/g, '<br/>');
                        if (tempMsg == "") {
                            tempMsg = "同步成功!";
                        }
                        SIE.Msg.showMessage(tempMsg);
                        view.reloadData();
                    }
                },
            });
        })
    }
});

